airbyte-integrations/connectors/source-pinterest/source_pinterest/manifest.yaml

```diff
@@ -0,0 +1,3190 @@
+
+  # Base stream
+  base_stream:
+    type: DeclarativeStream
+    retriever: "#/definitions/retriever"
+    primary_key: "id"
+
+  base_semi_incremental_stream:
+    $ref: "#/definitions/base_stream"
+    retriever:
+      $ref: "#/definitions/ad_account_based_retriever"
+      record_selector:
+        $ref: "#/definitions/selector"
+        record_filter:
+          type: RecordFilter
+          condition: "{{ record['updated_time'] >= stream_slice['start_time'] | int }}"
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: "updated_time"
+      datetime_format: "%s"
+      start_datetime: "{{ format_datetime(config.get('start_date'), '%s') if config.get('start_date') else 0 }}"
+
+  base_analytics_stream:
+    type: DeclarativeStream
+    primary_key: []
+    retriever: "#/definitions/analytics_retriever"
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: "{{ parameters.get('cursor_field', 'DATE') }}"
+      datetime_format: "%Y-%m-%d"
+      cursor_datetime_formats:
+        - "%s"
+      start_datetime: "{{ config.get('start_date') if config.get('start_date') and config.get('start_date') > day_delta(-89, format='%Y-%m-%d') else day_delta(-89, format='%Y-%m-%d') }}"
+      cursor_granularity: "P1D"
+      step: "P30D"
+      start_time_option:
+        type: RequestOption
+        field_name: "start_date"
+        inject_into: request_parameter
+      end_time_option:
+        type: RequestOption
+        field_name: "end_date"
+        inject_into: request_parameter
+
+  # Full refresh streams
+  boards_stream:
+    # Docs: https://developers.pinterest.com/docs/api/v5/#operation/boards/list
+    name: "boards"
+    $ref: "#/definitions/base_stream"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema: "#/definitions/boards_schema"
+    retriever:
```

Comment:
Can we move the `retriever` definition up to the `base_stream`. And then just override it as needed for streams like `audiences` which use the `#/definitions/ad_account_based_retriever`?

Reply:
Agree, updated